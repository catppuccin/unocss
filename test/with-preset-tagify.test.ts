import { test, expect } from 'vitest';
import { createGenerator, presetMini, presetTagify } from 'unocss';

import { extendCatppuccin } from '../dist';

test('presetMini + presetTagify without setting variant', async () => {
  const uno = createGenerator({
    presets: [
      presetMini({
        dark: 'media',
      }),
      presetTagify(),
    ],
    theme: {
      colors: extendCatppuccin(),
    },
  });

  const fixture = `
<text-mocha-flamingo>Hello world!</text-mocha-flamingo>
<text-frappe-pink>Hello world!</text-frappe-pink>
<text-macchiato-base>Hello world!</text-macchiato-base>
`;

  const { css } = await uno.generate(fixture, { preflights: false });

  expect(css).toMatchSnapshot();
});

test('presetMini + presetTagify with default latte variant', async () => {
  const uno = createGenerator({
    presets: [
      presetMini({
        dark: 'media',
      }),
      presetTagify(),
    ],
    theme: {
      colors: extendCatppuccin({
        variant: 'latte',
      }),
    },
  });

  const fixture = `
<text-flamingo>Hello world!</text-flamingo>
<text-pink>Hello world!</text-pink>
<text-base>Hello world!</text-base>
`;

  const { css } = await uno.generate(fixture, { preflights: false });

  expect(css).toMatchSnapshot();
});
