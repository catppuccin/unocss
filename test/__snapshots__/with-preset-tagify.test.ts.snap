// Vitest Snapshot v1

exports[`presetMini + presetTagify with default latte variant 1`] = `
"/* layer: default */
text-base{--un-text-opacity:1;color:rgba(239,241,245,var(--un-text-opacity));}
text-flamingo{--un-text-opacity:1;color:rgba(221,120,120,var(--un-text-opacity));}
text-pink{--un-text-opacity:1;color:rgba(234,118,203,var(--un-text-opacity));}"
`;

exports[`presetMini + presetTagify without setting variant 1`] = `
"/* layer: default */
text-frappe-pink{--un-text-opacity:1;color:rgba(244,184,228,var(--un-text-opacity));}
text-macchiato-base{--un-text-opacity:1;color:rgba(36,39,58,var(--un-text-opacity));}
text-mocha-flamingo{--un-text-opacity:1;color:rgba(242,205,205,var(--un-text-opacity));}"
`;
